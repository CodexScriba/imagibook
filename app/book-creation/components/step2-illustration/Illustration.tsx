import { IllustrationCarousel } from "./IllustrationCarousel";
import { IllustrationCharacterType } from "./IllustrationCharacterType";

export const Illustration: React.FC = () => {
    return (
        <div>
            <IllustrationCarousel />
            <IllustrationCharacterType />   
        </div>
    )
}