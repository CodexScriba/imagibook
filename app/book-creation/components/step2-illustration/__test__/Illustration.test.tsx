// Illustration.test.tsx

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // Provides custom matchers like toBeInTheDocument
import { Illustration } from '../Illustration';

// Mock the child components to isolate the test
jest.mock('../IllustrationCarousel', () => ({
  IllustrationCarousel: () => <div data-testid="illustration-carousel">Mocked Carousel</div>,
}));

jest.mock('../IllustrationCharacterType', () => ({
  IllustrationCharacterType: () => <div data-testid="illustration-character-type">Mocked Character Type</div>,
}));

describe('Illustration Component', () => {
  test('renders IllustrationCarousel and IllustrationCharacterType components', () => {
    // Render the Illustration component
    render(<Illustration />);

    // Assert that the mocked IllustrationCarousel is rendered
    const carouselElement = screen.getByTestId('illustration-carousel');
    expect(carouselElement).toBeInTheDocument();
    expect(carouselElement).toHaveTextContent('Mocked Carousel');

    // Assert that the mocked IllustrationCharacterType is rendered
    const characterTypeElement = screen.getByTestId('illustration-character-type');
    expect(characterTypeElement).toBeInTheDocument();
    expect(characterTypeElement).toHaveTextContent('Mocked Character Type');
  });

  test('matches the snapshot', () => {
    // Render the Illustration component
    const { asFragment } = render(<Illustration />);
    
    // Assert that the rendered output matches the snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
