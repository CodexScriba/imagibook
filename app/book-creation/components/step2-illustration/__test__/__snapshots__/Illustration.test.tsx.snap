// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Illustration Component matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="illustration-carousel"
    >
      Mocked Carousel
    </div>
    <div
      data-testid="illustration-character-type"
    >
      Mocked Character Type
    </div>
  </div>
</DocumentFragment>
`;
